runtime/commonMain/src/kotlinx/serialization/internal/ObjectSerializer.kt

```diff
@@ -13,8 +13,10 @@ import kotlinx.serialization.*
  * By default, a singleton is serialized as an empty structure, e.g. `{}` in JSON
  */
 @InternalSerializationApi
+@Suppress("DEPRECATED") // Used by compiler plugin
```

Comment:
There are probably some user-facing usages (e.g. serialize particular class instance as a singleton), so I suggest to leave this in internal api without deprecating

Reply:
Well, that's the point of **internal** API: we do not expect users to use it and thus may change it in the way we want unless it breaks binary compatibility